### ISSUE TITLE
Cypress.Cookies.defaults() doesn't work all the time

### ISSUE THREAD

**Issue by ghostling** [cypress-io/cypress#5177]
👋 I thought it would be clever to use `Cypress.Cookies.defaults({ whitelist: [cookies] });` in my `cypress/support/index.js` **instead** of using `Cypress.Cookies.preserveOnce(cookies)` in the `beforeEach` of every test suite.

However, I have two different ways of logging in depending on whether I'm running the tests against the local instance of the application or the live one in staging. The live one in staging works (cookies are set by `cy.request`), but the local one does not (cookies need to be set by `cy.setCookie`). Please see the abbreviated code snippet below.

While writing this issue, I noticed that this is not exactly reproducible 😞... Some of my colleagues run into this, some don't, and some eventually do. Wondering if there's some sort of global Cypress config hidden somewhere.

### Current behavior:

1. Use `Cypress.Cookies.defaults()` in `cypress/support/index.js` to save cookies across all tests.
2. Try to set one of the saved cookies manually in the `before` using `cy.setCookie()`
3. It will fail with an error message with:
```
TypeError: Cannot read property 'replace' of undefined

Because this error occurred during a 'before all' hook we are skipping the remaining tests in the current suite: 'Test Suite Name'
```
4. if we use the debugger, it will say:
```
{ [Error: Failed to parse or set cookie named "identity".] name: undefined }
```

### Desired behavior:
`Cypress.Cookies.defaults()` works with both `cy.request` and `cy.setCookie`

### Steps to reproduce: (app code and test code)

Not sure how much this helps because I can't share the actual "local" login pages
as they are internal only.
```js
// in cypress/support/index.js:
Cypress.Cookies.defaults({
  whitelist: ['identity', 'session-id'],
});

// in any test suite:
describe('some test', () => {
  before(() => {
    cy.loginViaApi();
  });
});

// loginViaApi is a custom command that does:
const loginViaApi = (email, password) => {
  if (ENV === 'local') {
    return localLoginViaApi(email, password);
  }

  // OK: Makes a request to login api, this sets cookies *correctly*
  return cy.request('/login');
};

// btw, we have to do this because the response has domain of `.www.local.example.com`
// which doesn't seem to automatically set cookies for...
// maybe related to https://github.com/cypress-io/cypress/issues/3221#issuecomment-499125203
import setCookie from 'set-cookie-parser';
const localLoginViaApi = (email, password) {
  cy.request('/localLoginEndpoint')
    .its('headers')
    .its('set-cookie')
    .then((setCookieHeader) => {
      const cookies = setCookie.parse(setCookieHeader);
      cookies.forEach((cookie) => {
        // HELP: this *fails* with
        cy.setCookie(cookie.name, cookie.value);
      });
    });
};
```

### Versions
- cypress 3.4.1
- fails both in cypress runner + headless mode

**Comment by alexeychikk** [comment]
I am trying to whitelist cookies in the `before()` like this:
```ts
before(() => {
  Cypress.Cookies.defaults({ whitelist: ["connect.sid"] });
});
```
It just **doesn't** work.

**Comment by alexeychikk** [comment]
So I figured out that this is due to SameSite updates in Chrome https://www.chromium.org/updates/same-site
which is not the case with Electron browser.

**Comment by jennifer-shehane** [comment]
@alexeychikk There are a few sameSite issues open, so we are working on handling this more correctly since it's newer. https://github.com/cypress-io/cypress/issues?q=is%3Aissue+is%3Aopen+sameSite

**Comment by jennifer-shehane** [comment]
I'm not able to reproduce this original issue. Is a bit difficult since the full example isn't actually provided and the origal issue says it only happens sometimes. My test:

```js
context('Cypress.Commands', () => {
  before(() => {
    cy.setCookie('fakeCookie', '123ABC')
  })
  
  it('works', () => {
    cy.visit('https://example.cypress.io/cypress-api')
    cy.getCookie('fakeCookie').its('value').should('eq', '123ABC')
  })

  it('still works', () => {
    cy.visit('https://example.cypress.io/cypress-api')
    cy.getCookie('fakeCookie').its('value').should('eq', '123ABC')
  })
})
```

Unfortunately we have to close this issue as there is not enough information to reproduce the problem. This does not mean that your issue is not happening - it just means that we do not have a path to move forward. 

I would also suggest trying the latest version of Cypress since there have been a lot of fixed since 3.4.1.

Please comment in this issue with a reproducible example and we will consider reopening the issue.
